From 88715bf9c47f8c08c87a9e82156839bd9ddeab4a Mon Sep 17 00:00:00 2001
From: Joe McGill <jmcgill@us.ibm.com>
Date: Thu, 29 Dec 2016 15:53:03 -0600
Subject: [PATCH v1 1013/1022] FIR updates -- pervasive/core/PPE

p9_obus_scom_address_fixes.H
  add OBUS IO PPE address constants

p9.cme.scan.initfile
  align EQ pervasive LFIR/XFIR settings with RAS XML docs

p9.core.scan.initfile
  align EC pervasive LFIR/XFIR settings with RAS XML docs

p9.core.scom.initfile
  adjust core FIR action settings for bits 1,12:13 to match RAS XML doc

p9_sbe_scominit.C
  mask PBA FIR bit 1 to match RAS XML doc
  initialize FBC/XBUS/OBUS PPE FIR registers

p9_sbe_common.C
  align non-EQ/EC pervasive LFIR/XFIR settings with RAS XML docs

Change-Id: Ifbc6a47eb2dbe268a7ea832e55986d46a1870420
---
 .../common/include/p9_obus_scom_addresses_fixes.H  | 30 ++++++-
 .../chips/p9/procedures/hwp/perv/p9_sbe_common.C   | 92 +++++++++++-----------
 2 files changed, 75 insertions(+), 47 deletions(-)

diff --git a/src/import/chips/p9/common/include/p9_obus_scom_addresses_fixes.H b/src/import/chips/p9/common/include/p9_obus_scom_addresses_fixes.H
index 7744643..eb982a3 100644
--- a/src/import/chips/p9/common/include/p9_obus_scom_addresses_fixes.H
+++ b/src/import/chips/p9/common/include/p9_obus_scom_addresses_fixes.H
@@ -5,7 +5,7 @@
 /*                                                                        */
 /* OpenPOWER HostBoot Project                                             */
 /*                                                                        */
-/* Contributors Listed Below - COPYRIGHT 2015,2016                        */
+/* Contributors Listed Below - COPYRIGHT 2015,2017                        */
 /* [+] International Business Machines Corp.                              */
 /*                                                                        */
 /*                                                                        */
@@ -61,4 +61,32 @@ REG64( OBUS_2_LL2_IOOL_CONTROL,
 REG64( OBUS_2_LL2_LL2_LL2_PB_IOOL_FIR_REG,
        RULL(0x0B010800), SH_UNT_OBUS_2, SH_ACS_SCOM );
 
+REG64( OBUS_0_IOPPE_PPE_FIR_ACTION0_REG,
+       RULL(0x09011046), SH_UNT_OBUS_0, SH_ACS_SCOM );
+REG64( OBUS_0_IOPPE_PPE_FIR_ACTION1_REG,
+       RULL(0x09011047), SH_UNT_OBUS_0, SH_ACS_SCOM );
+REG64( OBUS_0_IOPPE_PPE_FIR_MASK_REG,
+       RULL(0x09011043), SH_UNT_OBUS_0, SH_ACS_SCOM );
+
+REG64( OBUS_1_IOPPE_PPE_FIR_ACTION0_REG,
+       RULL(0x0A011046), SH_UNT_OBUS_1, SH_ACS_SCOM );
+REG64( OBUS_1_IOPPE_PPE_FIR_ACTION1_REG,
+       RULL(0x0A011047), SH_UNT_OBUS_1, SH_ACS_SCOM );
+REG64( OBUS_1_IOPPE_PPE_FIR_MASK_REG,
+       RULL(0x0A011043), SH_UNT_OBUS_1, SH_ACS_SCOM );
+
+REG64( OBUS_2_IOPPE_PPE_FIR_ACTION0_REG,
+       RULL(0x0B011046), SH_UNT_OBUS_2, SH_ACS_SCOM );
+REG64( OBUS_2_IOPPE_PPE_FIR_ACTION1_REG,
+       RULL(0x0B011047), SH_UNT_OBUS_2, SH_ACS_SCOM );
+REG64( OBUS_2_IOPPE_PPE_FIR_MASK_REG,
+       RULL(0x0B011043), SH_UNT_OBUS_2, SH_ACS_SCOM );
+
+REG64( OBUS_3_IOPPE_PPE_FIR_ACTION0_REG,
+       RULL(0x0C011046), SH_UNT_OBUS_3, SH_ACS_SCOM );
+REG64( OBUS_3_IOPPE_PPE_FIR_ACTION1_REG,
+       RULL(0x0C011047), SH_UNT_OBUS_3, SH_ACS_SCOM );
+REG64( OBUS_3_IOPPE_PPE_FIR_MASK_REG,
+       RULL(0x0C011043), SH_UNT_OBUS_3, SH_ACS_SCOM );
+
 #endif
diff --git a/src/import/chips/p9/procedures/hwp/perv/p9_sbe_common.C b/src/import/chips/p9/procedures/hwp/perv/p9_sbe_common.C
index 35d5bf9..648f529 100644
--- a/src/import/chips/p9/procedures/hwp/perv/p9_sbe_common.C
+++ b/src/import/chips/p9/procedures/hwp/perv/p9_sbe_common.C
@@ -5,7 +5,7 @@
 /*                                                                        */
 /* OpenPOWER HostBoot Project                                             */
 /*                                                                        */
-/* Contributors Listed Below - COPYRIGHT 2015,2016                        */
+/* Contributors Listed Below - COPYRIGHT 2015,2017                        */
 /* [+] International Business Machines Corp.                              */
 /*                                                                        */
 /*                                                                        */
@@ -76,60 +76,60 @@ const uint64_t PERV_LFIR_ACTION0[15] =
 
 const uint64_t PERV_LFIR_ACTION1[15] =
 {
-    0x8000000000000000ULL, // TP
-    0x8000000000000000ULL, // N0
-    0x8000000000000000ULL, // N1
-    0x8000000000000000ULL, // N2
-    0x8000000000000000ULL, // N3
-    0x8000000000000000ULL, // X
-    0x8000000000000000ULL, // MC0
-    0x8000000000000000ULL, // MC1
-    0x8000000000000000ULL, // OB0
-    0x8000000000000000ULL, // OB1
-    0x8000000000000000ULL, // OB2
-    0x8000000000000000ULL, // OB3
-    0x8000000000000000ULL, // PCI0
-    0x8000000000000000ULL, // PCI1
-    0x8000000000000000ULL  // PCI2
+    0xF000000000000000ULL, // TP
+    0xF000000000000000ULL, // N0
+    0xF000000000000000ULL, // N1
+    0xF000000000000000ULL, // N2
+    0xF000000000000000ULL, // N3
+    0xF000000000000000ULL, // X
+    0xF000000000000000ULL, // MC0
+    0xF000000000000000ULL, // MC1
+    0xF000000000000000ULL, // OB0
+    0xF000000000000000ULL, // OB1
+    0xF000000000000000ULL, // OB2
+    0xF000000000000000ULL, // OB3
+    0xF000000000000000ULL, // PCI0
+    0xF000000000000000ULL, // PCI1
+    0xF000000000000000ULL  // PCI2
 };
 
 const uint64_t PERV_LFIR_MASK[15] =
 {
-    0xFFFFFFFFFFC00000ULL, // TP
-    0xFFFFFFFFFFC00000ULL, // N0
-    0xFFFFFFFFFFC00000ULL, // N1
-    0xFFFFFFFFFFC00000ULL, // N2
-    0xFFFFFFFFFFC00000ULL, // N3
-    0xFFFFFFFFFFC00000ULL, // X
-    0xFFFFFFFFFFC00000ULL, // MC0
-    0xFFFFFFFFFFC00000ULL, // MC1
-    0xFFFFFFFFFFC00000ULL, // OB0
-    0xFFFFFFFFFFC00000ULL, // OB1
-    0xFFFFFFFFFFC00000ULL, // OB2
-    0xFFFFFFFFFFC00000ULL, // OB3
-    0xFFFFFFFFFFC00000ULL, // PCI0
-    0xFFFFFFFFFFC00000ULL, // PCI1
-    0xFFFFFFFFFFC00000ULL  // PCI2
+    0x0FFFBC2BFC400000ULL, // TP
+    0x0FFFFFFFFFC00000ULL, // N0
+    0x0FFFFFFFFFC00000ULL, // N1
+    0x0FFFFFFFFFC00000ULL, // N2
+    0x0FFFFFFF1FC00000ULL, // N3
+    0x0FFFFFFFFFC00000ULL, // X
+    0x0FFFFFFFFFC00000ULL, // MC0
+    0x0FFFFFFFFFC00000ULL, // MC1
+    0x0FFFFFFFFFC00000ULL, // OB0
+    0x0FFFFFFFFFC00000ULL, // OB1
+    0x0FFFFFFFFFC00000ULL, // OB2
+    0x0FFFFFFFFFC00000ULL, // OB3
+    0x0FFFFFFFFFC00000ULL, // PCI0
+    0x0FFFFFFFFFC00000ULL, // PCI1
+    0x0FFFFFFFFFC00000ULL  // PCI2
 };
 
 // chiplet XIR constants
 const uint64_t PERV_XFIR_MASK[15] =
 {
-    0x9FFFFFE000000000ULL, // TP
-    0x2007FFE000000000ULL, // N0
-    0x201FFFE000000000ULL, // N1
-    0x200FFFE000000000ULL, // N2
-    0x000007E000000000ULL, // N3
-    0x210FFFE000000000ULL, // X
-    0x20007FE000000000ULL, // MC0
-    0x20007FE000000000ULL, // MC1
-    0x29FFFFE000000000ULL, // OB0
-    0x29FFFFE000000000ULL, // OB1
-    0x29FFFFE000000000ULL, // OB2
-    0x29FFFFE000000000ULL, // OB3
-    0x21FFFFE000000000ULL, // PCI0
-    0x207FFFE000000000ULL, // PCI1
-    0x201FFFE000000000ULL  // PCI2
+    0x0000000000000000ULL, // TP
+    0x0000000000000000ULL, // N0
+    0x0000000000000000ULL, // N1
+    0x0000000000000000ULL, // N2
+    0x0000000000000000ULL, // N3
+    0x0000000000000000ULL, // X
+    0x0000000000000000ULL, // MC0
+    0x0000000000000000ULL, // MC1
+    0x0000000000000000ULL, // OB0
+    0x0000000000000000ULL, // OB1
+    0x0000000000000000ULL, // OB2
+    0x0000000000000000ULL, // OB3
+    0x0000000000000000ULL, // PCI0
+    0x0000000000000000ULL, // PCI1
+    0x0000000000000000ULL  // PCI2
 };
 
 /// @brief --For all chiplets exit flush
-- 
1.8.2.2

